app/views/discovered_hosts/welcome.html.erb

```diff
@@ -1,10 +1,16 @@
+<% content_for(:javascripts) do %>
+  <%= webpacked_plugins_js_for :'foreman_discovery' %>
+<% end %>
+<% content_for(:stylesheets) do %>
+  <%= webpacked_plugins_css_for :'foreman_discovery' %>
+<% end %>
+
 <% content_for(:title, _("Discovered Hosts")) %>
-<div class="blank-slate-pf">
-  <div class="blank-slate-pf-icon">
-    <%= icon_text("gears", "", :kind => "fa") %>
-  </div>
-  <h1><%= _('Discovered Hosts') %></h1>
-  <p><%= _("No discovered hosts found in this context.") %>
-  <%= _("This page shows discovered bare-metal or virtual nodes waiting to be provisioned.") %></p>
-  <p><%= link_to _('Learn more about this in the documentation.'), documentation_url("#{ForemanDiscovery::VERSION.scan(/\d+\.\d+/).first}/index.html", {:root_url => 'https://theforeman.org/plugins/foreman_discovery/'})%></p>
-</div>
+
+<% content_for(:content) do %>
+  <%= notifications %>
+  <div id="organization-id" data-id="<%= Organization.current.id if Organization.current %>" ></div>
+  <div id="user-id" data-id="<%= User.current.id if User.current %>" ></div>
+  <%= react_component('DiscoveredHosts',
```

Comment:
Maybe soon enough https://github.com/theforeman/foreman/pull/6936 will get merged and you could use the router registry instead :)

Reply:
Sure, I am also waiting for the core empty state PR to get merged! Will update this PR soon